src/foundry/client/pixi/layers/grid/hex.d.ts

```diff
@@ -1,118 +1,388 @@
-   * @param col - The column number
-   */
-  offsetToCube(row: number, col: number): { q: number; r: number; s: number };
+    w: number;
 
-  /**
-   * Convert a cube coordinate (q, r, s) into an offset coordinate (row, col).
-   * See https://www.redblobgames.com/grids/hexagons/ for reference
-   * Source code available https://www.redblobgames.com/grids/hexagons/codegen/output/lib-functions.js
-   * @param q - Cube coordinate 1
-   * @param r - Cube coordinate 2
-   * @param s - Cube coordinate 3
-   */
-  cubeToOffset(q: number, r: number, s: number): { row: number; col: number };
+    h: number;
 
-  /**
-   * Given a cursor position (x, y), obtain the cube coordinate hex (q, r, s) of the hex which contains it
-   * http://justinpombrio.net/programming/2020/04/28/pixel-to-hex.html
-   * @param x - The x-coordinate in pixels
-   * @param y - The y-coordinate in pixels
-   */
-  static pixelToCube(x: number, y: number): [q: number, r: number, s: number];
+    /**
+     * Compute the grid configuration from a provided type
+     * @param type - The grid type
+     * @param size - The grid size in pixels
+     */
+    static getConfig(type: number, size: number): HexGridConfiguration;
 
-  /**
-   * Measure the distance in hexagons between two cube coordinates
-   */
-  static cubeDistance(a: { q: number; r: number; s: number }, b: { q: number; r: number; s: number }): number;
+    /**
+     * Special border polygons for different token sizes.
+     */
+    static POINTY_HEX_BORDERS: Record<number, PointArray[]>;
+
+    /**
+     * Special border polygons for different token sizes.
+     */
+    static FLAT_HEX_BORDERS: Record<number, PointArray[]>;
+
+    /**
+     * A matrix of x and y offsets which is multiplied by the width/height vector to get pointy-top polygon coordinates
+     */
+    static get pointyHexPoints(): PointArray[];
+
+    /**
+     * A matrix of x and y offsets which is multiplied by the width/height vector to get flat-top polygon coordinates
+     */
+    static get flatHexPoints(): PointArray[];
+
+    /**
+     * An array of the points which define a hexagon for this grid shape
+     */
+    get hexPoints(): PointArray[];
+
+    override draw(options?: BaseGrid.DrawOptions | undefined): this;
+
+    /**
+     * A convenience method for getting all the polygon points relative to a top-left [x,y] coordinate pair
+     * @param x      - The top-left x-coordinate
+     * @param y      - The top-right y-coordinate
+     * @param w      - An optional polygon width
+     * @param h      - An optional polygon height
+     * @param points - An optional list of polygon points.
+     */
+    getPolygon(x: number, y: number, w?: number, h?: number, points?: PointArray[]): PointArray[];
+
+    /**
+     * Draw the grid lines.
+     * @param preview - Override settings used in place of those saved to the scene data.
+     * @internal
+     */
+    protected _drawGrid(options?: BaseGrid.DrawOptions | undefined): PIXI.Graphics;
```

Comment:
I think `| undefined` is redundant here. I know I've been talking a lot about `InexactPartial` etc. and adding `| undefined`  to optional stuff but that's only for properties on object-like types. The reason for it is `exactOptionalPropertyTypes`, a compiler flag that makes `x?: number` and `x?: number | undefined` slightly different. Notably in the first `x: undefined` isn't allowed.